bot/utils/services.py

```diff
@@ -1,50 +1,105 @@
+import dataclasses
+import datetime
+import json
+import typing
 from typing import Optional
 
+import aiohttp
 from aiohttp import ClientConnectorError
+from async_rediscache import RedisCache
 
 import bot
 from bot.constants import URLs
 from bot.log import get_logger
 
 log = get_logger(__name__)
+T = typing.TypeVar("T")
 
 FAILED_REQUEST_ATTEMPTS = 3
 
+UNFURL_CACHE = RedisCache(namespace="UnfurledRedirects")
+CACHE_LENGTH = datetime.timedelta(days=1)
 
-async def send_to_paste_service(contents: str, *, extension: str = "") -> Optional[str]:
+
+@dataclasses.dataclass()
```

Comment:
I'm guessing this object is supposed to be immutable after being created, since we don't update the return. As such we could add `eq=True, frozen=True` to the dataclass call here. This will improve performance and make it hashable/allow equivalence checks.

Reply:
Yeah sounds reasonable. 80abaef2